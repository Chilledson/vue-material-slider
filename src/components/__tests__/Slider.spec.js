import Slider from '../slider';
import { mount } from '@vue/test-utils';

const mountComponent = (options = {}) => {
  return mount(Slider, options);
}

let wrapper = null

afterEach(() => {
  wrapper.destroy()
})

it('renders correctly', () => {
  wrapper = mountComponent();
  expect(wrapper.html()).toMatchSnapshot();
});

it('renders vertically', () => {
  wrapper = mountComponent({ 
    propsData: {
      vertical: true
    }
  });
  expect(wrapper.html()).toMatchSnapshot();
});

it('renders inverted', () => {
  wrapper = mountComponent({ 
    propsData: {
      inverted: true
    }
  });
  expect(wrapper.html()).toMatchSnapshot();
});

it('renders disabled', () => {
  wrapper = mountComponent({ 
    propsData: {
      disabled: true
    }
  });
  expect(wrapper.html()).toMatchSnapshot();
});

it('should change value on key press', () => {
  wrapper = mountComponent();

  wrapper.trigger('keydown.up');

  expect(wrapper.vm.localValue).toBe(1);

  wrapper.trigger('keydown.down');

  expect(wrapper.vm.localValue).toBe(0);

  wrapper.trigger('keydown.pageup');

  expect(wrapper.vm.localValue).toBe(10);

  wrapper.trigger('keydown.pagedown');
  
  expect(wrapper.vm.localValue).toBe(0);

  wrapper.trigger('keydown.end');
  
  expect(wrapper.vm.localValue).toBe(100);
  
  wrapper.trigger('keydown.home');

  expect(wrapper.vm.localValue).toBe(0);

  wrapper.trigger('keydown.right');

  expect(wrapper.vm.localValue).toBe(1);

  wrapper.trigger('keydown.left');

  expect(wrapper.vm.localValue).toBe(0);
});

it('should increment with a given step size', () => {
  wrapper = mountComponent({
    propsData: {
      stepSize: 10
    }
  });

  wrapper.trigger('keydown.up');
  expect(wrapper.vm.localValue).toBe(10);
})

it('should call change on drag', () => {
  wrapper = mountComponent();

  wrapper.trigger('mousedown');
  expect(wrapper.vm.localValue).toBe(0);
})

it('should emit change event slide start', () => {
  wrapper = mountComponent();

  wrapper.trigger('mousedown');
  expect(wrapper.emitted().change).toBeDefined();
})

it('should emit change event slide end', () => {
  wrapper = mountComponent();

  wrapper.trigger('touchstart');
  wrapper.trigger('touchend');
  expect(wrapper.emitted().change).toBeDefined();
})

