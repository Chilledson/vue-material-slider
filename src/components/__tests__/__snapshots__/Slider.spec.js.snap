// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div tabindex="0" class="slider slider-horizontal slider-min-value" style="user-select: none; webkit-user-drag: none; -webkit-tap-highlight-color: rgba(0, 0, 0, 0);">
  <div class="slider-wrapper">
    <div class="slider-track-wrapper">
      <div class="slider-track-background" style="transform: translateX(7px) scale3d(1, 1, 1);"></div>
      <div class="slider-track-fill" style="transform: translateX(-7px) scale3d(0, 1, 1);"></div>
    </div>
    <div class="slider-ticks-container">
      <div class="slider-ticks"></div>
    </div>
    <div class="slider-thumb-container" style="transform: translateX(-100%);">
      <div class="slider-focus-ring"></div>
      <div class="slider-thumb"></div>
      <div class="slider-thumb-label"><span class="slider-thumb-label-text">0</span></div>
    </div>
  </div>
</div>
`;

exports[`renders disabled 1`] = `
<div tabindex="0" class="slider slider-disabled slider-horizontal slider-min-value" style="user-select: none; webkit-user-drag: none; -webkit-tap-highlight-color: rgba(0, 0, 0, 0);">
  <div class="slider-wrapper">
    <div class="slider-track-wrapper">
      <div class="slider-track-background" style="transform: translateX(7px) scale3d(1, 1, 1);"></div>
      <div class="slider-track-fill" style="transform: translateX(-7px) scale3d(0, 1, 1);"></div>
    </div>
    <div class="slider-ticks-container">
      <div class="slider-ticks"></div>
    </div>
    <div class="slider-thumb-container" style="transform: translateX(-100%);">
      <div class="slider-focus-ring"></div>
      <div class="slider-thumb"></div>
      <div class="slider-thumb-label"><span class="slider-thumb-label-text">0</span></div>
    </div>
  </div>
</div>
`;

exports[`renders inverted 1`] = `
<div tabindex="0" class="slider slider-horizontal slider-min-value" inverted="true" style="user-select: none; webkit-user-drag: none; -webkit-tap-highlight-color: rgba(0, 0, 0, 0);">
  <div class="slider-wrapper">
    <div class="slider-track-wrapper">
      <div class="slider-track-background" style="transform: translateX(7px) scale3d(1, 1, 1);"></div>
      <div class="slider-track-fill" style="transform: translateX(-7px) scale3d(0, 1, 1);"></div>
    </div>
    <div class="slider-ticks-container">
      <div class="slider-ticks"></div>
    </div>
    <div class="slider-thumb-container" style="transform: translateX(-100%);">
      <div class="slider-focus-ring"></div>
      <div class="slider-thumb"></div>
      <div class="slider-thumb-label"><span class="slider-thumb-label-text">0</span></div>
    </div>
  </div>
</div>
`;

exports[`renders vertically 1`] = `
<div tabindex="0" class="slider slider-vertical slider-axis-inverted slider-min-value" style="user-select: none; webkit-user-drag: none; -webkit-tap-highlight-color: rgba(0, 0, 0, 0);">
  <div class="slider-wrapper">
    <div class="slider-track-wrapper">
      <div class="slider-track-background" style="transform: translateY(-7px) scale3d(1, 1, 1);"></div>
      <div class="slider-track-fill" style="transform: translateY(7px) scale3d(1, 0, 1);"></div>
    </div>
    <div class="slider-ticks-container">
      <div class="slider-ticks"></div>
    </div>
    <div class="slider-thumb-container" style="transform: translateY(-0%);">
      <div class="slider-focus-ring"></div>
      <div class="slider-thumb"></div>
      <div class="slider-thumb-label"><span class="slider-thumb-label-text">0</span></div>
    </div>
  </div>
</div>
`;
